### PR TITLE
feat(ops): fast calls for Wasm

### PR DESCRIPTION
**Pull request by littledivy**
This PR introduces Wasm ops. These calls are optimized for entry from Wasm land. 

The `#[op(wasm)]` attribute is opt-in. 

Last parameter `Option<&mut [u8]>` is the memory slice of the Wasm module *when entered from a Fast API call*. Otherwise, the user is expected to implement logic to obtain the memory if `None`

```rust
#[op(wasm)]
pub fn op_args_get(
  offset: i32,
  buffer_offset: i32,
  memory: Option<&mut [u8]>,
) {
  // ...
}
```

### CODE REVIEW

File: ops/attrs.rs [hunk 1]
```diff
@@ -20,18 +21,27 @@ impl Parse for Attributes {
...
         return Err(Error::new(
           input.span(),
-          "invalid attribute, expected one of: unstable, v8, fast, deferred",
+          "invalid attribute, expected one of: unstable, v8, fast, deferred, wasm",
         ));
       }
     }
+
+    let is_wasm = vars.contains(&"wasm");
+    let mut must_be_fast = vars.contains(&"fast");
```
Comment (aapoalas) [inline ops/attrs.rs:33]: `is_wasm || vars.contains(&"fast");`?

File: core/examples/wasm.rs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  let memory_object = unsafe {
+    transmute::<NonNull<v8::WasmMemoryObject>, v8::Local<v8::WasmMemoryObject>>(
+      *global,
+    )
+  };
+  let backing_store = memory_object.buffer().get_backing_store();
+  let ptr = backing_store.data().unwrap().as_ptr() as *mut u8;
+  let len = backing_store.byte_length();
+  // SAFETY: `ptr` is a valid pointer to `len` bytes.
+  unsafe { std::slice::from_raw_parts_mut(ptr, len) }
```
Comment (aapoalas) [inline core/examples/wasm.rs:31]: Is it possible that ptr is a null pointer? eg. Wasm module created with 0 length memory buffer?

Comment (bartlomieju) [reply core/examples/wasm.rs:31]: @littledivy what's the answer here?

Comment (littledivy) [reply core/examples/wasm.rs:31]: Quick test shows that it is not nullptr for an empty memory buffer.

Comment (littledivy) [reply core/examples/wasm.rs:31]: Maybe it can be...idk...this is just an example

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**